packages/datatrak-web-server/src/types.ts

```diff
@@ -3,8 +3,12 @@
  *  Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
  */
 
-import { ModelRegistry, EntityModel } from '@tupaia/database';
+import { ModelRegistry, EntityModel, EntityType as BaseEntityType } from '@tupaia/database';
+import { Model } from '@tupaia/server-boilerplate';
+import { Entity } from '@tupaia/types';
+
+export type EntityType = BaseEntityType & Entity;
 
 export interface DatatrakWebServerModelRegistry extends ModelRegistry {
-  readonly entity: EntityModel;
+  readonly entity: Model<EntityModel, Entity, EntityType>;
```

Comment:
This is probably what we want to be doing when defining ModelRegistry types. By wrapping in the `Model` type, we get access to the various methods defined there (`findById` is the one we want in this context).

Reply:
Oh I didn't know about that Generic, thanks for the info. @EMcQ-BES you probably already know about this but in case not take a look